src/components/TablePagination.tsx

```diff
@@ -0,0 +1,193 @@
+import {
+  Button,
+  Icon,
+  Input,
+  Select,
+  usePagination,
+} from "@canonical/react-components";
+import React, {
+  ChangeEvent,
+  Children,
+  FC,
+  HTMLAttributes,
+  PropsWithChildren,
+  ReactElement,
+  ReactNode,
+  RefObject,
+  cloneElement,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import classnames from "classnames";
+
+const figureSmallScreen = (descriptionRef: RefObject<HTMLDivElement>) => {
+  const descriptionElement = descriptionRef.current;
+  if (!descriptionElement) {
+    return true;
+  }
+  return descriptionElement.getBoundingClientRect().width < 230;
+};
+
+const renderChildren = (
+  children: ReactNode,
+  dataPropOverride: string,
+  data: unknown[],
+) => {
+  return Children.map(children, (child) => {
+    return cloneElement(child as ReactElement, {
+      [dataPropOverride]: data,
+    });
+  });
+};
+
+const DEFAULT_PAGE_LIMITS = [50, 100, 200];
+const generatePagingOptions = (pageLimits: number[]) => {
+  return pageLimits.map((limit) => ({ value: limit, label: `${limit}/page` }));
+};
+
+interface Props {
+  data: unknown[];
+  itemName?: string;
+  containerClass?: string;
+  displayTitle?: ReactNode;
+  renderBelow?: boolean;
+  dataPropOverride?: string;
+  pageLimits?: number[];
+  dataFiltered?: boolean;
+}
+
+const TablePagination: FC<
+  PropsWithChildren<Props> & HTMLAttributes<HTMLDivElement>
+> = ({
+  data,
+  containerClass,
+  itemName = "item",
+  displayTitle,
+  renderBelow,
+  dataPropOverride = "rows",
```

Comment:
Maybe you can find a better name, though I don't have a good idea right away. Maybe keep as is.

Reply:
we can think on this one a bit, can't think of a better name either

Reply:
maybe `propForDataOverride`?

Reply:
Seems too long and not concise enough. Maybe `dataForwardProp`?

Reply:
that's nice, thanks! :+1: